ACHNBrowserUI/Packages/Backend/Sources/Backend/Environments/UserCollection.swift

```diff
@@ -132,29 +132,33 @@ public class UserCollection: ObservableObject {
         return added
     }
     
-    public func variantIn(item: Item, variant: Variant) -> Bool {
-        guard let filename = item.filename else {
-            return false
-        }
-        return variants[filename]?.contains(variant) == true
-    }
-    
     public func toggleVariant(item: Item, variant: Variant) -> Bool {
         guard let filename = item.filename else {
             return false
         }
+        let isAdded: Bool
         if variants[filename]?.contains(variant) == true {
             variants[filename]?.removeAll(where: { $0 == variant })
             save()
-            return false
+            isAdded = false
         } else {
             if variants[filename] == nil {
                 variants[filename] = []
             }
             variants[filename]?.append(variant)
             save()
-            return true
+            isAdded = true
         }
+
+        if !items.contains(item) && variants.completionStatus(for: item) != .unstarted {
+            // We add the item to the liked items if at least one of its variant is added
+            toggleItem(item: item)
+        } else if items.contains(item) && variants.completionStatus(for: item) == .unstarted {
+            // We remove the item from the liked items if there is none of its variants liked
+            toggleItem(item: item)
```

Comment:
Same

Reply:
Done, I'm also wondering if at the end of the day `@discardableResult` could come handy for the function. As it's not used in a lot of places, I think that's fine for now.